presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AssignUniqueId.java

```diff
@@ -72,4 +74,11 @@ public Symbol getIdColumn()
     {
         return visitor.visitAssignUniqueId(this, context);
     }
+
+    @Override
+    public PlanNode replaceChildren(List<PlanNode> newChildren)
```

Comment:
what about renaming this to `replaceSources`? You have `getSources()` but `replaceChildren`, it sounds like they were referencing to two different things.